node/src/effect.rs

```diff
@@ -798,6 +798,20 @@ impl<REv> EffectBuilder<REv> {
         .await
     }
 
+    /// Requests the switch block _for_ the given era ID, ie. the switch block at the era before
```

Comment:
I guess the switch block for the given era is a _block_ at the era before. It can't be a switch blocks for both eras.

Reply:
Note that the first part says "for", and the second part says "at" :wink: A switch block _at_ era N is a switch block _for_ era N+1.

Reply:
It _is_ confusing, isn't it? Maybe we should call it "key block" after all? :innocent:

Reply:
Alright, let me rename that :stuck_out_tongue: